ENIGMAsystem/SHELL/Graphics_Systems/Direct3D11/DX11textures_impl.h

```diff
@@ -0,0 +1,39 @@
+/** Copyright (C) 2008-2013 Josh Ventura
+*** Copyright (C) 2013,2019 Robert B. Colton
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#ifdef INCLUDED_FROM_SHELLMAIN
+#  error This file includes non-ENIGMA STL headers and should not be included from SHELLmain.
+#endif
+
+#ifndef ENIGMA_DX11_TEXTURES_IMPL_H
+#define ENIGMA_DX11_TEXTURES_IMPL_H
+
+#include "Direct3D11Headers.h"
+#include "Graphics_Systems/General/GStextures_impl.h"
+
+namespace enigma {
+
+struct DX11Texture : Texture {

```

Comment:
Can probably extract a template here (for D3D9/11), but still, whatever.

Reply:
That's true, I didn't think of that. Once we have a generic class to cast to (e.g, `enigma::Texture) that doesn't mean we can't put the derived in `General/` and template it. There's only one problem though, DX11 is different from the other three. DX11's holds two peers actually, the peer texture itself and a shader resource view to be able to access the texture in a shader. I haven't exactly looked into this yet but it could be true that the shader resource view does not honestly belong where I've put it.